options.go

```diff
@@ -275,3 +276,11 @@ func WithTestErrorContains(with string) Option {
 		return nil
 	}
 }
+
+// WithRemoveUnusedCredentials, if set to true, indicates that unused credentials should be removed
+func WithRemoveUnusedCredentials(with bool) Option {
+	return func(o *Options) error {
+		o.WithRemoveUnusedCredentials = with
+		return nil
+	}
+}
```

Comment:
Nit: Is it expected that this option is supported by all storage implementations? From an interface point of view, it seems weird that only `Load` accepts this. I might want to do this at any point of interacting with the storage, no?

Alternatively, why do we only sometimes want to remove unused credentials? Can't we do it unconditionally in all implementations of storage?

Reply:
@jefferai I know you didn't want to unconditionally remove unused creds, but now that I'm looking I'm not sure that we would ever be in that situation unless it was a set of creds the worker had already discarded from rotation. Is there something I'm missing?

Reply:
To be honest, I'm not even sure how we'd quantify "unused" creds, _especially_ in the context of a load. What is the perceived flow here?

Reply:
What we've run into is a worker rotates credentials [previous, current] and has a rotation failure trying to rotate from [A,B] to [B,C]. Worker view on creds is:`[A,B]`, controller is `[B,C]`

The worker then tries to rotate to `[B,D]`. Boundary's storage still has B, but stored under previous. The root of this bug is that we'd populate node credentials with the key id for B but would use key data for C.

We can detect that C is unused by noticing that B is stored as the previous key- we wouldn't be storing C as current unless the worker informed us it intended to use C as its new current, so if we're seeing the worker using a key we have stored as previous, it must mean that the worker didn't receive notification that C was successfully stored and discarded it.  I couldn't find a situation where we would load a previous NodeInformation if a current had been negotiated.